packages/loot-core/src/server/accounts/transaction-rules.ts

```diff
@@ -274,8 +275,20 @@ function onApplySync(oldValues, newValues) {
 }
 
 // Runner
-export async function runRules(trans) {
-  let finalTrans = await prepareTransactionForRules({ ...trans });
+export async function runRules(
+  trans,
+  accounts: Map<string, AccountEntity> = null,
```

Comment:
Is there a reason we're passing this in instead of always generating as below?

Reply:
because there are two places that runRules is executed inside a for loop:
sync.ts:530 and sync.ts:682
but to not poison this code "/packages/loot-core/src/server/rules/app.ts:144" with a db call, I let it live inside runRules and also outside